### PR TITLE
[SIL-67] Details screen - network layer

### PR DESCRIPTION
**Pull request by Krzysiudan**
https://silentus-au.atlassian.net/jira/software/projects/SIL/boards/1?selectedIssue=SIL-67

Changes: 
- implement requests for the reviews and requested reviews (both are needed)
- implemented tests for that requests
- added auto date parsing (with java.time, for that added desugaring library - including newer API on lower API versions) https://developer.android.com/studio/write/java8-support#library-desugaring
- added parsing Github error response into our `WebException.UnknownError` with a valid message. 

UI:  There are no visual changes in the app.

### CODE REVIEW

File: .github/workflows/run-code-quality-check.yml [hunk 1]
```diff
@@ -17,7 +17,11 @@ concurrency:
   group: ${{ github.ref }}-${{ github.workflow }}
   cancel-in-progress: true
 
-permissions: write-all
+permissions:
+  checks: write
+  contents: write
+  statuses: write
+  pull-requests: write
```
Comment (Krzysiudan) [inline .github/workflows/run-code-quality-check.yml:24]: **Info** Fixing linter issue

File: .github/workflows/run-unit-test.yml [hunk 2]
```diff
@@ -4,8 +4,10 @@ on:
   pull_request:
   push:
     branches:
-      - 'develop'
-      - 'main'
+      - "develop"
+      - "main"
+
+permissions: read-all
```
Comment (Krzysiudan) [inline .github/workflows/run-unit-test.yml:10]: **Info** Fixing linter issue

File: .jscpd.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "threshold": 0,
   "reporters": ["html", "console", "xml"],
-  "ignore": ["**/__snapshots__/**"],
+  "ignore": ["**/__snapshots__/**", "**/src/test/java/**"],
```
Comment (Krzysiudan) [inline .jscpd.json:4]: **Info** Fixing linter issue - ignoring test source files from checking for duplicated code (json schemas were duplicated).

File: app/build.gradle [hunk 4]
```diff
@@ -54,6 +55,9 @@ android {
 }
 
 dependencies {
+    //Desugaring for use of java.time in api lower then 26
+    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:2.0.2'
+
```
Comment (Krzysiudan) [inline app/build.gradle:60]: **Info** This enables to use java.time in api's below 26.

File: app/src/main/java/com/appunite/loudius/domain/GitHubPullRequestsRepository.kt [hunk 5]
```diff
@@ -1,10 +1,43 @@
...
+    PullRequestRepository {
+    override suspend fun getPullRequestsForUser(author: String): Result<PullRequestsResponse> =
         remoteDataSource.getPullRequestsForUser(author)
+
+    override suspend fun getReviews(
+        owner: String,
+        repo: String,
+        pullRequestNumber: String,
+    ): Result<List<Review>> =
+        remoteDataSource.getReviews(owner, repo, pullRequestNumber, "TODO ACCESS TOKEN")
```
Comment (Krzysiudan) [inline app/src/main/java/com/appunite/loudius/domain/GitHubPullRequestsRepository.kt:35]: **Info** Wojtek is working on adding an access token to the requests by interceptor due to that I will leave it as it is for now.

File: .github/workflows/run-code-quality-check.yml (hunk 1 above)
Comment (nowakweronika) [reply .github/workflows/run-code-quality-check.yml:24]: Suggestion: I think, we can do a seperate PRs in the future with changes like this.

File: app/src/main/java/com/appunite/loudius/network/services/GithubPullRequestsService.kt [hunk 6]
```diff
@@ -13,4 +16,20 @@ interface GithubPullRequestsService {
         @Query("page") page: Int = 0,
         @Query("per_page") perPage: Int = 100,
     ): PullRequestsResponse
+
+    @GET("/repos/{owner}/{repo}/pulls/{pull_number}/requested_reviewers")
+    suspend fun getReviewers(
```
Comment (kezc) [inline app/src/main/java/com/appunite/loudius/network/services/GithubPullRequestsService.kt:21]: question: is this network implementation not used in the screen yet?

**Review by kezc — APPROVED** [review]
i have nothing to say, lgtm

**Review by nowakweronika — APPROVED** [review]
lgtm

**Pull request merged**